clang/lib/AST/ASTImporter.cpp

```diff
@@ -3008,17 +3008,30 @@ ExpectedDecl ASTNodeImporter::VisitRecordDecl(RecordDecl *D) {
         // declaration), ASTContext does not check it.
         SmallVector<Decl *, 2> Redecls =
             getCanonicalForwardRedeclChain(D2CXX);
-        const Type *FrontTy =
-            cast<CXXRecordDecl>(Redecls.front())->getTypeForDecl();
+
+        // FIXME: This is an LLDB-specific fix for crashes
+        // related to rdar://120862220, where the redeclaration
+        // chain can contain decls that don't have a TypeForDecl.
+        // In such cases we want to grab the TypeForDecl from some
+        // other declaration on the chain.
+        const Type *FrontTy = nullptr;
+        for (auto const& Redecl : Redecls) {
+          FrontTy = cast<CXXRecordDecl>(Redecl)->getTypeForDecl();
+          if (FrontTy)
+            break;
+        }
+
+        assert (FrontTy != nullptr);
```

Comment:
I think for LLVM code an assert message is required (and code formatting is not standard).

Reply:
fixed